RFS/src/test/java/com/rfs/common/RestClientTest.java

```diff
@@ -0,0 +1,107 @@
+package com.rfs.common;
+
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader;
+import org.opensearch.migrations.testutils.HttpRequestFirstLine;
+import org.opensearch.migrations.testutils.SimpleNettyHttpServer;
+import org.opensearch.migrations.testutils.SimpleHttpResponse;
+
+class RestClientTest {
+
+    private InMemoryMetricReader metricReader;
+
+    @BeforeEach
+    public void setup() {
+        metricReader = InMemoryMetricReader.create();
+        final var meterProvider = SdkMeterProvider.builder()
+            .registerMetricReader(metricReader)
+            .build();
+
+        OpenTelemetrySdk.builder()
+            .setMeterProvider(meterProvider)
+            .buildAndRegisterGlobal();
+    }
+
+    @Test
+    public void testGetEmitsInstrumentation() throws Exception{
+        try (var testServer = SimpleNettyHttpServer.makeServer(false, null, this::makeResponseContext)) {
+            var restClient = new RestClient(new ConnectionDetails("http://localhost:" + testServer.port, null, null));
+            restClient.postAsync("/", "empty").block();
+            restClient.getAsync("/").block();
+        }
+
+        // Thread.sleep(200);
+        var allMetricData = metricReader.collectAllMetrics();
+
+        allMetricData.forEach(m -> {
+            System.out.println(m);
+        });
+
+        for (var kvp : Map.of(
+                "POST", new int[]{133, 66},
+                "GET", new int[]{139, 66},
+                "", new int[]{272, 132}).entrySet()) {
+            long bytesSent = allMetricData.stream().filter(md -> md.getName().startsWith("writeBytes"))
+                    .reduce((a, b) -> b).get().getLongSumData().getPoints()
+                    .stream()
+                    .filter(pd -> pd.getAttributes().asMap().values().stream().noneMatch(attrVal -> attrVal.equals(kvp.getKey())))
+                    .mapToLong(pd -> pd.getValue())
+                    .sum();
+            Assertions.assertEquals(kvp.getValue()[0], bytesSent);
+
+
+            long bytesRead = allMetricData.stream().filter(md -> md.getName().startsWith("readBytes"))
+                    .reduce((a, b) -> b).get().getLongSumData().getPoints()
+                    .stream()
+                    .filter(pd -> pd.getAttributes().asMap().values().stream().noneMatch(attrVal -> attrVal.equals(kvp.getKey())))
+                    .mapToLong(pd -> pd.getValue())
+                    .sum();
+            Assertions.assertEquals(kvp.getValue()[1], bytesRead);
+        }
+        // var finishedSpans = rootContext.instrumentationBundle.getFinishedSpans().stream()
```

Comment:
Doesn't seem correct to validate spans in this scenario, as there really isn't an interesting 'context' where things are occurring since the client is declared in the test case.

I think the scenario still needs coverage, need to drill in more.  Not sure how important _detailed_ span traces are for RFS, curious for your thoughts @gregschohn